### DISCUSSION TITLE
reason behind 'tap_" prefix when creating a tap device for local bridging

### DISCUSSION THREAD

**Discussion by shakibamoshiri** [SoftEtherVPN/SoftEtherVPN#1794] [General]
When creating a tap device (or maybe is called vLAN) SE adds "tap_" name as a prefix to each interface. e.g. 

```bash
tap_site1 ...
tap_site2 ...
```

I am wondering what does this "tap_" prefix do and **is it safe to remove this "tap_" prefix** ?

```C
// Generate TUN interface name
void GenerateTunName(char *name, char *prefix, char *tun_name, size_t tun_name_len)
{
	char instance_name_lower[MAX_SIZE];

	// Generate the device name
	StrCpy(instance_name_lower, sizeof(instance_name_lower), name);
	Trim(instance_name_lower);
	StrLower(instance_name_lower);
	Format(tun_name, tun_name_len, "%s_%s", prefix, instance_name_lower);

	tun_name[15] = 0;
}
```

So having this style

```bash
site1 ...
site2 ...
```

**Comment by chipitsine** [comment]
"tap" is a part of naming convention, i.e. you can enumerate interfaces and find local bridges

**Reply by shakibamoshiri** [reply]
I am wondering what benefit it will have by adding this "tap_" prefix ? Just for the sake of convention has been added?
I never saw an interface with "tap_" on Linux since we can check the interface type using "iprotue2" tools.

**Reply by chipitsine** [reply]
benefits of naming conventions are - you can build a list of network interfaces and separate "tap_*", i.e. created for certain purpose.

**Reply by shakibamoshiri** [reply]
> benefits of naming conventions are - you can build a list of network interfaces and separate "tap_*", i.e. created for certain purpose.

I think type detection though naming convention is not the way of identifying devices.
it could like this
 - `brg_docker0`
 - `???_wg0`
 - `tun_openvpn0`
 - `tap_hub0`

Why not ?
 - `docker0`
 - `wg0`
 - `tun0`
 - `hub0` 

and check their type using other tools

```bash
ip --json -d link show  | jq  -r  '.[].linkinfo.info_data.type'
```

**Anyway, is it safe to remove this "tap_" prefix ?**  
I think modifying this function `GenerateTunName` will do it. Returning a name without 'tap_" prefix.

**Reply by shipitsin-ilia** [reply]
there might be users who use that naming convention. you can break things for them.

**Reply by shakibamoshiri** [reply]
> there might be users who use that naming convention. you can break things for them.

I am not going to do anything to the source code here, it is just for myself on my local machine. Is it safe to remove it ?

**Reply by shipitsin-ilia** [reply]
sure
